src/clj/domain/spec.clj

```diff
@@ -52,10 +53,21 @@
 (s/def :question-set/questions (s/coll-of ::question))
 
 
+;; Added specs :question-set/start, question-set/end, :questions/passing-score
+;; These specs came into the code as requirements but werent supplied
+;; with the necessary specifications.
+(s/def :question-set/start inst?)
+(s/def :question-set/end  inst?)
```

Comment:
Maybe we should use your util function to check if it is a date object here. I think it would be beneficial, as we also check it when the `time/start-before-end?` function is used.

Reply:
That's a good objection, I also think that way but the is-date-object? sadly cannot be applied with
a generator to exercise the question-set spec.
inst? has the same behavior as is-date-object? concerning Java.util.Date instances but has
the advantage to be compatible with spec/exercise.
I don't know if the latter argument is enough though to justify the
obfuscation of our spec by inst?.

Reply:
I've read the documentation and looked into the function and now I understand what this function actually does.
But by god I did not get it because of the doc string:

> Return true if x satisfies Inst

And what a name for a function. But ok. If we can use it for test generation then let's use it.

Reply:
But let's maybe remove the comments. I don't see them being helpful in the future. And let's maybe add a comment explaining that `inst?` checks if the value is an instance of java.util.Date

Reply:
Addes some comments for clarification.